templates/todo/projects/nodejs-mongo/.repo/terraform/infra/main.tf

```diff
@@ -28,7 +28,110 @@ module "applicationinsights" {
   location         = var.location
   rg_name          = azurerm_resource_group.rg.name
   environment_name = var.environment_name
-  workspace_id     = azurerm_log_analytics_workspace.workspace.id
+  workspace_id     = module.loganalytics.LOGANALYTICS_WORKSPACE_ID
   tags             = azurerm_resource_group.rg.tags
   resource_token   = local.resource_token
 }
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy log analytics
+# ------------------------------------------------------------------------------------------------------
+module "loganalytics" {
+  source         = "../../../../../../common/infra/terraform/loganalytics"
+  location       = var.location
+  rg_name        = azurerm_resource_group.rg.name
+  tags           = azurerm_resource_group.rg.tags
+  resource_token = local.resource_token
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy key vault
+# ------------------------------------------------------------------------------------------------------
+module "keyvault" {
+  source                   = "../../../../../../common/infra/terraform/keyvault"
+  location                 = var.location
+  principal_id             = var.principal_id
+  rg_name                  = azurerm_resource_group.rg.name
+  tags                     = azurerm_resource_group.rg.tags
+  resource_token           = local.resource_token
+  access_policy_object_ids = [module.appserviceapi.APP_OBJECT_ID]
+  secrets = [
+    {
+      name  = "AZURE-COSMOS-CONNECTION-STRING"
+      value = module.cosmos.AZURE_COSMOS_CONNECTION_STRING_KEY
+    }
+  ]
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy cosmos
+# ------------------------------------------------------------------------------------------------------
+module "cosmos" {
+  source         = "../../../../../../common/infra/terraform/cosmos"

```

Comment:
Please move to /core folder like we have with bicep

Reply:
moved